tests/base/test_cyl3D.py

```diff
@@ -0,0 +1,702 @@
+from __future__ import print_function
+import unittest
+import numpy as np
+import sympy
+from sympy.abc import r, t, z
+
+import discretize
+from discretize import Tests, utils
+
+np.random.seed(16)
+
+TOL = 1e-1
+
+
+class TestCyl3DGeometries(unittest.TestCase):
+    def setUp(self):
+        hx = utils.meshTensor([(1, 1)])
+        htheta = utils.meshTensor([(1., 4)])
+        htheta = htheta * 2*np.pi / htheta.sum()
+        hz = hx
+
+        self.mesh = discretize.CylMesh([hx, htheta, hz])
+
+    def test_areas(self):
+        area = self.mesh.area
+        self.assertTrue(self.mesh.nF == len(area))
+        self.assertTrue(
+            area[:self.mesh.vnF[0]].sum() == 2*np.pi*self.mesh.hx*self.mesh.hz
+        )
+        self.assertTrue(
+            np.all(
+                area[self.mesh.vnF[0]:self.mesh.vnF[1]] ==
+                self.mesh.hx*self.mesh.hz
+            )
+        )
+        self.assertTrue(
+            np.all(
+                area[self.mesh.vnF[:2].sum():] ==
+                np.pi * self.mesh.hx**2 / self.mesh.nCy
+            )
+        )
+
+    def test_edges(self):
+        edge = self.mesh.edge
+        self.assertTrue(self.mesh.nE == len(edge))
+        self.assertTrue(
+            np.all(edge[:self.mesh.vnF[0]] == self.mesh.hx)
+        )
+        self.assertTrue(
+            np.all(
+                self.mesh.edge[self.mesh.vnE[0]:self.mesh.vnE[:2].sum()] ==
+                np.kron(np.ones(self.mesh.nCz+1), self.mesh.hx*self.mesh.hy)
+            )
+        )
+        self.assertTrue(
+            np.all(
+                self.mesh.edge[self.mesh.vnE[0]:self.mesh.vnE[:2].sum()] ==
+                np.kron(np.ones(self.mesh.nCz+1), self.mesh.hx*self.mesh.hy)
+            )
+        )
+        self.assertTrue(
+            np.all(
+                self.mesh.edge[self.mesh.vnE[:2].sum():] ==
+                np.kron(self.mesh.hz, np.ones(self.mesh.nCy+1))
+            )
+        )
+
+    def test_vol(self):
+        self.assertTrue(
+            self.mesh.vol.sum() == np.pi * self.mesh.hx ** 2 * self.mesh.hz
+        )
+        self.assertTrue(
+            np.all(
+                self.mesh.vol ==
+                np.pi * self.mesh.hx ** 2 * self.mesh.hz / self.mesh.nCy
+            )
+        )
+
+# ----------------------- Test Grids and Counting --------------------------- #
+
+
+class Cyl3DGrid(unittest.TestCase):
```

Comment:
I would appreciate another set of eyes here. I tried to pick a simple example that demonstrated how things are counted. See what you think